File: Prueba y error/ejercicios de semana santa java/three java/Library/PrimeraLey.java
package Library;

//  la Primera Ley de Newton (Inercia)
public class PrimeraLey extends LeyNewton {

    public PrimeraLey() {
        super("Primera Ley de Newton - Inercia");
    }

    @Override
    public String explicarLey() {
        return "Si F = 0, el objeto mantiene su estado (reposo o movimiento uniforme).";
    }

    @Override
    public String calcular(double... valores) {
        if (valores.length < 1) {
            return "Error: se necesita la fuerza para determinar el estado del objeto.";
        }
        double fuerza = valores[0];
        if (fuerza == 0) {
            return "El objeto mantiene su estado (reposo o movimiento uniforme).";
        } else {
            return "El objeto cambiará su estado debido a la fuerza aplicada.";
        }
    }
}

File: Prueba y error/ejercicios de semana santa php/Thre php/Library/PrimeraLey.php
<?php
namespace Library;

require_once 'LeyNewton.php';

class PrimeraLey extends LeyNewton {

    public function describir(): string {
        return "Primera Ley de Newton - Inercia";
    }

    public function ejecutar(): void {
        echo "===== " . $this->describir() . " =====<br>";
        echo "Un objeto en reposo permanecerá en reposo, y un objeto en movimiento continuará en movimiento a velocidad constante a menos que una fuerza externa actúe sobre él.<br>";
    }
}